kong/plugins/ip-restriction/handler.lua

```diff
@@ -1,22 +1,45 @@
+local lrucache = require "resty.lrucache"
 local ipmatcher = require "resty.ipmatcher"
 local kong_meta = require "kong.meta"
 
 
-local ngx = ngx
+local ngx_var = ngx.var
 local kong = kong
 local error = error
 
 
+local IPMATCHER_COUNT = 512
+local IPMATCHER_TTL   = 300
```

Comment:
Good idea, I will change it to 3600.

Reply:
This cache worries me. It could cause the following scenario:

1. Admin configures the plugin to allow all IPs to access
2. `192.168.10.5` access the resource, the decision to allow is cached for 3600 seconds
3. Admin uses Admin API to disallow access from `192.168.10.5`
4. `192.168.10.5` can still access because the result from 2 is still cached

Is this an acceptable behavior? My feeling is that the cache is altering the plugin behavior and causing cache poisoning which seems to be wrong.

Reply:
We only cache the ipmacher list, not the allow/deny result.

If admin API changed the plugin's config, the key of cache is also changed too, so we will get a new ipmatcher list.

Reply:
I see, thanks for clarifying!